[u]int64_t parameters; decimal->number cast removed in favor of original textual representation

parihaaraka opened the pull request.

Commit:
[u]int64_t parameters; decimal->number cast removed in favor of original textual representation

parihaaraka commented:
> Patch breaks tests:

Because client-side decimal->number cast is a faulty one. Take a look at [Qt's NumericalPrecisionPolicy](https://doc.qt.io/qt-5/qsql.html#NumericalPrecisionPolicy-enum). If a user has no choice then we should preserve original precision.

ligurio commented:
>> Patch breaks tests:
> Because client-side decimal->number cast is a faulty one.

Patch cannot be merged if it makes CI red. Please fix tests.

parihaaraka committed:
new `connection.dec_cast` option ('s' - string, 'n' - number, default is 'n')

parihaaraka committed:
optional pg decimal to tnt decimal cast

parihaaraka commented:
> It changes :execute() arguments order. So the old code will need to be updated anyway.

No, this is not connection's execute() - it's internal one. Everything is absolutely backward compatible. Previous behaviour is the default one.

Totktonada commented:
NB: link #26, check whether we should copy the static buffer before passing it to libpq (in binding parameters handling)

parihaaraka committed:
multiple [u]int64_t parameters usage fix

parihaaraka commented:
@Totktonada Ъ?

Totktonada commented:
Sorry, I'm on finishing my current quarter tasks. Hopefully I'll end on this week.

ligurio committed:
Set a type for PostreSQL decimal numbers [TO SQUASH]

[REMOVE BEFORE MERGE] Changes proposed in fixup commit: documentation,
regression tests, code style.

Patch adds support of PostgreSQL decimal numbers - it is possible to
choose a Lua/Tarantool type that will be used for casting  values with
NUMERIC type [1]. By default Lua "number" type is used, the same
behaviour was used before changes. Additionally user could choose two
other types: Lua "string" and Tarantool's decimal type [2], [3].

Fixes #26
Fixes #25

1. https://www.postgresql.org/docs/current/datatype-oid.html
2. https://www.tarantool.io/en/doc/latest/reference/reference_lua/decimal/
3. https://www.tarantool.io/en/doc/latest/book/box/data_model/#data-types

Co-authored-by: Sergey Bronnikov <sergeyb@tarantool.org>

ligurio committed:
Fix fallthrough compiler warnings

1. https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wimplicit-fallthrough

parihaaraka commented:
@Totktonada Ъ?
